### PR TITLE
docs: add instructions for connecting edge host to wifi

### PR DESCRIPTION
**Pull request by lennessyy**
## Describe the Change

This PR adds instructions on how to connect an Intel NUC edge host to Wifi. 

## Review Changes

💻 [Connect to Wifi](https://deploy-preview-2092--docs-spectrocloud.netlify.app/clusters/edge/networking/connect-wifi)

- [Apply site user data](https://deploy-preview-2092--docs-spectrocloud.netlify.app/clusters/edge/site-deployment/site-installation/site-user-data)

🎫 [Jira Ticket](https://spectrocloud.atlassian.net/jira/software/c/projects/PE/issues/PE-3511?filter=myopenissues)

### CODE REVIEW

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    apt-get install iputils-ping -y && \
+    mkdir /var/lib/wpa
+   ```
+
+   :::
+
+3. Follow the rest of the steps in the guide Build Edge Artifact. When finished, identify the
+   **palette-edge-installer.iso** file in the **build** folder.
+4. Plug in your bootable device and flash it with the ISO image.
+5. Plug the bootable device into the Edge host and power on the Edge host. When the Edge host powers on, open the boot
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:71]: Should menu be Menu? I'll defer to you Lenny.

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+   lines.
+
+   ```
+   RUN apt-get update && apt-get install wpasupplicant && \
+       mkdir /var/lib/wpa
+   ```
+
+   :::tip
+
+   Consider including some other lightweight network utilities in the OS image as well that can help you troubleshoot or
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:55]:
```suggestion
   Consider including some other lightweight network utilities in the OS image that can help you troubleshoot or
```

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+   set up network configurations alternatively if needed. For example, replace the RUN statement with the following to
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md]:
```suggestion
   set up network configurations alternatively if needed. For example, replace the RUN statement with the following commands to
```

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+   ```
+   RUN apt-get update && apt-get install wpasupplicant -y && \
+    apt-get update && apt-get install network-manager -y && \
+    apt-get install iputils-ping -y && \
+    mkdir /var/lib/wpa
+   ```
+
+   :::
+
+3. Follow the rest of the steps in the guide Build Edge Artifact. When finished, identify the
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:68]: Missing link for Build Edge Artifact guide?

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+   **palette-edge-installer.iso** file in the **build** folder.
+4. Plug in your bootable device and flash it with the ISO image.
+5. Plug the bootable device into the Edge host and power on the Edge host. When the Edge host powers on, open the boot
+   menu by pressing F10, and select the bootable device as your boot volume.
+6. Wait for installation to finish. The Edge host will reboot by default upon completion unless you specify a different
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:73]:
```suggestion
6. Wait for the installation to finish. The Edge host will reboot by default upon completion unless you specify a different
```

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+### Validate
+
+You can validate that the Edge host is ready for the site installation by powering on the Edge host and simulating a
+site deployment. The simulation process will require you to complete the installation process and reset the device after
+the validation.
+
+## Supply Site-Specific User Data
+
+It's likely that your Wi-Fi network name and passwords are site-specific, which means you won't be able to include the
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:84]:
```suggestion
It is likely that your Wi-Fi network name and passwords are site-specific, which means you won't be able to include the
```

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+- A bootable device such as a USB drive, or a Preboot Execution Environment (PXE) server.
+- `mkisofs`, or `genisoimage`, or similar ISO management software.
+  - `cdrtools` or `wodim` for Windows.
+- An Edge host that has already been prepared for installation.
+
+### Procedure
+
+1. Create a file named `user-data`, and include the following content. Replace `network-name` and `network-password`
+   with the name and password of your Wifi network.
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:98]:
```suggestion
   with the name and password of your Wi-Fi network.
```

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+            if [ -n "$wireless_interface" ]; then
+                wpa_passphrase <network-name> <network-password> | tee /var/lib/wpa/wpa_supplicant.conf
+                wpa_supplicant -B -c /var/lib/wpa/wpa_supplicant.conf -i $wireless_interface
+                dhclient $wireless_interface
+            else
+                echo "No wireless network interface found."
+            fi
+   ```
+
+2. Follow the rest of the steps in [Apply Site User Data](../site-deployment/site-installation/site-user-data.md).
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:131]:
```suggestion
2. Follow the rest of the steps in the [Apply Site User Data](../site-deployment/site-installation/site-user-data.md) section.
```

File: docs/docs-content/clusters/edge/networking/connect-wifi.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+### Validate
+
+1. Power up the Edge host and wait for it to boot to the registration screen. Do not connect the Edge host with a wired
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/networking/connect-wifi.md:135]: Maybe a call out box for **Do not connect the Edge host with a wired**?

File: docs/docs-content/clusters/edge/site-deployment/site-installation/site-user-data.md [hunk 10]
```diff
@@ -20,45 +20,83 @@ to a bootable device, such as a USB stick.
...
 
-<br />
+    ```shell
+    touch user-data
+    ```
 
-```shell
-touch user-data
-```
+    For example, you can include the following content in the **user-data** file to connect your Edge host to wifi. This
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/site-user-data.md:36]:
```suggestion
    For example, you can include the following content in the **user-data** file to connect your Edge host to Wi-Fi. This
```

File: docs/docs-content/clusters/edge/site-deployment/site-installation/site-user-data.md [hunk 11]
```diff
@@ -72,13 +110,16 @@ This generates an ISO file called site-user-data.iso in the current directory. <
...
 
-<br />
-
 ## Validate
 
 You can validate that the ISO image is not corrupted by attempting to flash a bootable device. Most software that
 creates a bootable device will validate the ISO image before the flash process.
 
+If you have SSH access, you can also SSH into the Edge host and locate your **user-data** file in either `/oem` or
+`/run/stylus`. The site-specific user-data is named **user-data** while the original user data file is named something
```
Comment (caroldelwing) [inline docs/docs-content/clusters/edge/site-deployment/site-installation/site-user-data.md:119]:
```suggestion
`/run/stylus`. The site-specific user data is named **user-data** while the original user data file is named something
```

**Review by caroldelwing — APPROVED** [review]
Really nice how-to!

**Pull request merged**